src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -87,35 +90,25 @@ of this software and associated documentation files (the "Software"), to deal
      */
     @SuppressFBWarnings(value="DC_DOUBLECHECK")
     public static DatadogClient getInstance(String url, String logIntakeUrl, Secret apiKey){
-        if(enableValidations){
-            if (url == null || url.isEmpty()) {
-                logger.severe("Datadog Target URL is not set properly");
-                return null;
-            }
-            if (apiKey == null || Secret.toString(apiKey).isEmpty()){
-                logger.severe("Datadog API Key is not set properly");
-                return null;
-            }
-            if (logIntakeUrl == null || logIntakeUrl.isEmpty()){
-                logger.warning("Datadog Log Intake URL is not set properly");
-            }
-        }
-
-        if(instance == null){
-            synchronized (DatadogHttpClient.class) {
-                if(instance == null){
-                    instance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+        // If the configuration has not changed, return the current instance without validation
```

Comment:
If the configuration is invalid, each time the code will call `getInstance`, new validation requests will be made.
As an improvement to limit the number of requests I would add a time condition "Is the configuration invalid and was validation not run in the past 5min, then run validations".

Reply:
> If the configuration is invalid, each time the code will call getInstance, new validation requests will be made.

Are you referring to the same invalid parameters or different ones?  

I also realized from this diff I removed the synchronized  block, which should be added back in.

Reply:
> Are you referring to the same invalid parameters or different ones?

Same parameters, on init lastInstance is null so if the config is incorrect we'll always run validation

Reply:
We initialize last instance before validation so it's not null with incorrect parameters.